odoo/models.py

```diff
@@ -3568,10 +3568,12 @@ def write(self, vals):
         protected_fields = map(self._fields.get, new_vals)
         with self.env.protecting(protected_fields, self):
             # write old-style fields with (low-level) method _write
-            if old_vals:
+            if old_vals or new_vals:
```

Comment:
It does check access rules as a side effect. In the case where `old_vals` is empty, it will perform an update of fields `write_date` and `write_uid`, and check access rules at the same time.

Reply:
Oh right, it does have an effect if `log_access` is enabled for the model. It does deserve a comment then ;-)